File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/PagedCouponResponseDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Data class representing a response containing a paginated list of coupon data.
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class PagedCouponResponseDTO {
    Long lastFetchTime; // Epoch milliseconds
    Long totalCoupon;
    Integer totalPage;
    Integer currentPage;
    List<CouponSummaryDTO> courses;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/PagedCouponResponseDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * Data class representing a response containing a paginated list of coupon data.
 */
data class PagedCouponResponseDTO(
    // Epoch milliseconds
    val lastFetchTime: Long? = null,
    val totalCoupon: Long? = null,
    val totalPage: Int? = null,
    val currentPage: Int? = null,
    val courses: List<CouponSummaryDTO>? = null,
)
